lib/scss_lint/linter/merge_equal_selector.rb

```diff
@@ -0,0 +1,60 @@
+module SCSSLint
+  class Linter::MergeEqualSelector < Linter
+    include LinterRegistry
+
+    def check_node(node)
+      visits = Set.new
+      node.children.each do |child_node|
+
+        equal_node = find_equal_node child_node, node.children
+        next unless equal_node
+        next if visits.include? [child_node, equal_node] or
+                visits.include? [equal_node, child_node]
+        visits << [child_node, equal_node]
+
+        add_lint child_node.line,
+        "Merge rule `#{node_rule(child_node)}` on line #{child_node.line} with rule " \
+        "on line #{equal_node.line}" do
+          node.children.delete equal_node
+
+          equal_node.children.each do |ec_node|
+            if ec_node.is_a? Sass::Tree::PropNode
+              index = child_node.children.rindex{ |n| n.is_a? Sass::Tree::PropNode } + 1
+              index ||= child_node.children.index{ |n| n.is_a? Sass::Tree::RuleNode }
+            else
+              index = -1
+            end
+            child_node.children.insert index, ec_node
+          end
+        end
+      end
+
+      yield # Continue linting children
+    end
+
+    alias_method :visit_root, :check_node
+    alias_method :visit_rule, :check_node
+
+  protected
```

Comment:
`private` would be preferred here since we don't intend to have other instances of this class refer to its methods.